packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -102,7 +104,96 @@ exports.sourceNodes = async (
     return
   }
 
-  const drupalFetchActivity = reporter.activityTimer(`Fetch data from Drupal`)
+  fastBuilds = fastBuilds || false
+  if (fastBuilds) {
+    let lastFetched = 0
+    if (
+      store.getState().status.plugins &&
+      store.getState().status.plugins[`gatsby-source-drupal`] &&
+      store.getState().status.plugins[`gatsby-source-drupal`].lastFetched
+    ) {
+      lastFetched = store.getState().status.plugins[`gatsby-source-drupal`]
+        .lastFetched
+    }
+
+    const drupalFetchIncrementalActivity = reporter.activityTimer(
+      `Fetch incremental changes from Drupal`
+    )
+    let requireFullRebuild = false
+
+    drupalFetchIncrementalActivity.start()
+
+    try {
+      // Hit fastbuilds endpoint with the lastFetched date.
+      const data = await axios.get(
+        `${baseUrl}/gatsby-fastbuilds/sync/${lastFetched}`,
+        {
+          auth: basicAuth,
+          headers,
+          params,
+        }
+      )
+
+      if (data.data.status === -1) {
+        reporter.info(`Unable to pull incremental data changes from Drupal`)
+        setPluginStatus({ lastFetched: data.data.timestamp })
+        requireFullRebuild = true
+      } else {
+        // Touch nodes so they are not garbage collected by Gatsby.
+        const existingNodes = getNodes().filter(
+          n => n.internal.owner === `gatsby-source-drupal`
+        )
+        existingNodes.forEach(n => touchNode({ nodeId: n.id }))
```

Comment:
This should probably be one loop. Especially considering the `existingNodes` is not used anywhere else. In that case it shouldn't `filter` (as that creates a new array) but rather do `forEach` which only loops.
```suggestion
        getNodes().forEach(node => {
          if (node.internal.owner === `gatsby-source-drupal`) {
            touchNode({ nodeId: node.id })
          }
        })
```

Reply:
Could even be this, although I'm not a fan because this feels like it reads worse:

```suggestion
        getNodes().forEach(({id: nodeId, internal: {owner}}) => {
          if (owner === `gatsby-source-drupal`) {
            touchNode({ nodeId })
          }
        })
```